system-tests/lib/cre/types.go

```diff
@@ -579,11 +811,13 @@ func (t *DonTopology) ToDonMetadata() []*DonMetadata {
 	return metadata
 }
 
+// CapabilitiesAwareNodeSet is the serialized form that declares nodesets in a topology.
 type CapabilitiesAwareNodeSet struct {
 	*ns.Input
-	Capabilities         []string          `toml:"capabilities"` // global capabilities that have no chain-specific configuration (like cron, web-api-target, web-api-trigger, etc.)
-	DONTypes             []string          `toml:"don_types"`
-	SupportedChains      []uint64          `toml:"supported_chains"`     // chain IDs that the DON supports, empty means all chains
+	Capabilities    []string `toml:"capabilities"` // global capabilities that have no chain-specific configuration (like cron, web-api-target, web-api-trigger, etc.)
+	DONTypes        []string `toml:"don_types"`
+	SupportedChains []uint64 `toml:"supported_chains"` // chain IDs that the DON supports, empty means all chains
+	// TODO separate out bootstrap as a concept rather than index
```

Comment:
I do agree, but maybe we should have core and edge structs and adapters between them? In this case only the edge struct would be built from TOML. Otherwise we might be forced to do what we do with `Capabilities`/`ChainCapabilities` and `ComputedCapabilities`, i.e. have both the index field and a bootstrap node field.